code/controllers/subsystem/timer.dm

```diff
@@ -236,7 +236,7 @@ SUBSYSTEM_DEF(timer)
 	if (flags & TIMER_UNIQUE)
 		SStimer.hashes[hash] = src
 	if (flags & TIMER_STOPPABLE)
-		SStimer.timer_id_dict["timerid[id]"] = src
+		SStimer.timer_id_dict["timerid"+num2text(id,8)] = src
```

Comment:
Just move the id = nextid++ line to here, 

and then either move the name line to below these ifs, or move this if to above the name line, as it references the id, if any.